linker/Mono.Linker.Steps/AddBypassNGenStep.cs

```diff
@@ -0,0 +1,121 @@
+﻿//
+// AddBypassNGenStep.cs
+//
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Collections.Generic;
+using Mono.Cecil.Cil;
+
+namespace Mono.Linker.Steps {
+
+	public class AddBypassNGenStep : BaseStep{
+
+		AssemblyDefinition coreLibAssembly;
+		CustomAttribute bypassNGenAttribute;
+		
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (Annotations.GetAction (assembly) == AssemblyAction.AddBypassNGen) {
+				coreLibAssembly = Context.Resolve (assembly.MainModule.TypeSystem.CoreLibrary);
```

Comment:
You could check for null no to repeat this all the time

Reply:
No, I can't. The reason is that assembly1.MainModule.TypeSystem.CoreLibrary and assembly2.MainModule.TypeSystem.CoreLibrary may not be the same. For example, assembly1 may reference System.Runtime.dll while assembly2 may reference System.Private.CoreLib.dll. This may happen in the context of the same app. We actually have such scenarios with ASP.NET apps.

In general, currently there may be up to 4 corelib dlls: https://github.com/mono/cecil/blob/master/Mono.Cecil/TypeSystem.cs#L282

I considered 3 solutions for this:

1. Figure out the assembly that defines System.Object (i.e., System.Private.CoreLib.dll), add BypassNGenAttribute there, and add missing assembly references to it (e.g., to assembly1 in the example above). This may cause problems for assemblies that will reference multiple core assemblies.

2. Figure out the assembly that defines System.Object, add BypassNGenAttribute there, and add type forwarders to it from other core assemblies (e.g., from System.Runtime.dll in the example above).

3. Add BypassNGenAttribute to each of the corelib assemblies that needs it. 

The code currently implements option 3. That's simpler than option 2 and should be fine since we have at most 4 core assemblies.

Reply:
Yeah, not sure it's all worth the trouble and injecting private attribute to assemblies which need it wouldn't be enough